harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -210,6 +364,26 @@ def _set_data_loaders(self) -> None:
                     )
                 self.validation_loader = validation_data
 
+    def _run(self):
+        min_checkpoint_period = self.env.experiment_config.get_min_checkpoint_period()
+        min_validation_period = self.env.experiment_config.get_min_validation_period()
+
+        for op in self.core_context.searcher.operations(core.SearcherMode.ChiefOnly):
+            searcher_unit = self.core_context.searcher.get_configured_units()
+            train_unit = None
+            if searcher_unit == core.Unit.EPOCHS:
+                train_unit = Epoch(op.length)
+                min_checkpoint_period = max(1, min_checkpoint_period.get("epochs"))
+                min_validation_period = max(1, min_validation_period.get("epochs"))
+
+            elif searcher_unit == core.Unit.BATCHES:
+                train_unit = Batch(op.length)
+                min_checkpoint_period = max(1, min_checkpoint_period.get("batches"))
+                min_validation_period = max(1, min_validation_period.get("batches"))
```

Comment:
... or every batch!  lmao that's so much validation

Reply:
lol. the checkpoint/validation period logic is now more in line with current behavior.